xdsl/transforms/function_transformations.py

```diff
@@ -0,0 +1,47 @@
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, func
+from xdsl.dialects.builtin import ArrayAttr, DictionaryAttr, StringAttr
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class ArgNamesToArgAttrsPass(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: func.FuncOp, rewriter: PatternRewriter, /):
+        arg_attrs = list(op.arg_attrs or (len(op.args) * [DictionaryAttr({})]))
```

Comment:
Would it not be worth doing something like this?
```suggestion
    def match_and_rewrite(self, op: func.FuncOp, rewriter: PatternRewriter, /):
        if not any(arg.name_hint for arg in op.args):
            return
        arg_attrs = list(op.arg_attrs or (len(op.args) * [DictionaryAttr({})]))
```
This way there's no intermediate thing created if there's nothing to do

Reply:
At the cost of an extra pass over the args? I actually didn't think that's what we wanted.

Reply:
Done as suggested.